src/lib/storage/table.cpp

```diff
@@ -203,6 +216,54 @@ TableType Table::get_type() const {
   }
 }
 
+void Table::set_partitioning_and_clear(std::shared_ptr<AbstractPartitionSchema> partition_schema) {
+  _partition_schema = std::move(partition_schema);
+  _chunks.clear();
+}
+
+void Table::apply_partitioning(std::shared_ptr<AbstractPartitionSchema> partition_schema) {
+  if (row_count() > 0) {
+    throw std::runtime_error("Unable to create partitioning on non-empty table");
+  }
+  _partition_schema = partition_schema;
+  _create_initial_chunks(static_cast<PartitionID>(partition_schema->partition_count()));
+}
+
+void Table::create_hash_partitioning(const ColumnID column_id, const HashFunction hash_function,
```

Comment:
Do we need to expose both `apply` and `create`? Also, a "create" method would have to return something.

Reply:
@mrzzzrm - You said that there were too many methods in the table. I agree, and it is mostly because of these methods that don't really add too much. If we remove the combined "create/apply" methods, I think we are good.

Reply:
`set_partitioning_and_clear` seems like unclean design as well. Same for a `get_mutable_partition_schema` that is just available in debug mode.